src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -421,7 +421,12 @@ const KeyCondition::AtomMap KeyCondition::atom_map
             if (value.getType() != Field::Types::String)
                 return false;
 
-            String prefix = extractFixedPrefixFromRegularExpression(value.get<const String &>());
+            const String & expression = value.get<const String &>();
+            // This optimization can't process alternation - this would require a comprehensive parsing of regular expression.
```

Comment:
I thought a bit 
- whether the check should rather be moved into `extractFixedPrefixFromRegularExpression()` but I think it is okay to not do so
- whether `extractFixedPrefixFromRegularExpression()` could be replaced by the more sophisticated regex parsing in `OptimizedRegularExpression::analyze()`, but again no. `::analyze()` searches the *longest* substring and that's not necessarily the prefix.

Therefore, we can merge the fix. Thanks.